pyfar/dsp/filter/gammatone.py

```diff
@@ -0,0 +1,529 @@
+    """
+
+    def __init__(self, freq_range, resolution=1, reference_frequency=1000,
+                 delay=0.004, sampling_rate=44100):
+
+        # check input (remaining checks done in erb_frequencies)
+        freq_range = np.asarray(freq_range)
+        if np.any(freq_range < 0) or np.any(freq_range > sampling_rate / 2):
+            raise ValueError(("Values in freq_range must be between 0 Hz and "
+                              "sampling_rate/2"))
+        if delay <= 0:
+            raise ValueError("The delay must be larger than zero")
+        if resolution <= 0:
+            raise ValueError("The resolution must be larger than zero")
+
+        # store user values
+        self._freq_range = freq_range
+        self._resolution = resolution
+        self._reference_frequency = reference_frequency
+        self._delay = delay
+        self._sampling_rate = sampling_rate
+
+        # compute center frequencies
+        self._frequencies = erb_frequencies(
+            freq_range, resolution, reference_frequency)
+        # compute filter coefficients
+        self._coefficients, self._normalizations = self._get_coefficients()
+        # initialize the internal filter state
+        self._state = None
+        # compute the filter delay, phase factor, and gains
+        # (required for the re-synthesis)
+        self._delays, self._phase_factors = \
+            self._get_delays_and_phase_factors()
+        self._gains = self._get_gains()
+
+    def __repr__(self):
+        """Nice string representation of class instances"""
+        return (f"Reconstructing Gammatone filter bank with {self.n_bands} "
+                f"bands between {self.freq_range[0]} and {self.freq_range[1]} "
+                f"Hz spaced by {self.resolution} ERB units @ "
+                f"{self.sampling_rate} Hz sampling rate")
+
+    @property
+    def freq_range(self):
+        """Get the frequency range of the filter bank in Hz"""
+        return self._freq_range
+
+    @property
+    def resolution(self):
+        """Get the resolution of the filter bank in ERB units"""
+        return self._resolution
+
+    @property
+    def reference_frequency(self):
+        """Get the reference frequency of the filter bank in Hz"""
+        return self._reference_frequency
+
+    @property
+    def frequencies(self):
+        """Get the center frequencies of the Gammatone filters in Hz"""
+        return self._frequencies
+
+    @property
+    def n_bands(self):
+        """Get the number of bands in the filter bank"""
+        return len(self.frequencies)
+
+    @property
+    def delay(self):
+        """Get the desired delay of the filter bank in seconds"""
+        return self._delay
+
+    @property
+    def sampling_rate(self):
+        """Get the sampling rate of the filter bank in Hz"""
+        return self._sampling_rate
+
+    @property
+    def coefficients(self):
+        """
+        Get the filter coefficients a as in Eq. (7) in Hohmann 2002 per band.
```

Comment:
Generating the docs locally at my computer cuts off everything after `Eg.` in the attributes table, could you recheck, please? Probably this is not an issue of this PR...

Reply:
yes, I also noted that and my lazy mind blamed Sphinx and Inspid for it :)